envconfig.go

```diff
@@ -19,7 +19,8 @@ import (
 // ErrInvalidSpecification indicates that a specification is of the wrong type.
 var ErrInvalidSpecification = errors.New("specification must be a struct pointer")
 
-var gatherRegexp = regexp.MustCompile("([^A-Z]+|[A-Z][^A-Z]+|[A-Z]+)")
+var gatherRegexp = regexp.MustCompile("([^A-Z]+|[A-Z]+[^A-Z]+|[A-Z][^A-Z]+|[A-Z]+)")
```

Comment:
Isn't it the same as `"([^A-Z]+|[A-Z]+[^A-Z]+|[A-Z]+)"`?

Reply:
indeed, thanks

Reply:
You should probably squash commit ;)

Reply:
sure, before merging i'm gonna squash it